src/case_gather/xml_parser.py

```diff
@@ -1,181 +1,138 @@
+            'althingi_status': getStatus(details_soup),
+            'subject_names': getSubjectNames(details_soup),
+            'althingi_link': case['althingi_link'],
+            'case_creator_names': getCaseCreatorNames(details_soup)
+        }
+        cases_data.append(case)
+
+    return cases_data
+
+def collectCases(soup):
     """
-    Function to yield from get_subjects
-    Format of output is [subject_id, group_name, subject_name,
-                                        subject_description, case_numbers]
+        Collects cases from the soup
+        Interesting information:
+            - Case number
+            - Case name
+            - Case type ( single letter identifier)
+            - Althingi_link (link to case on athingi.is)
     """
-    for i in range(1, 100):
-        output = get_subject(i)
-        if output is not None:
-            yield output
-        else:
-            pass
 
-
-def get_cases(parliament_session_number):
+    cases = []
+    for case in getAllCases(soup):
+        number = getCaseNumberFromCase(case)
+        name = getCaseName(case)
+        case_type = getCaseType(case)
+        althingi_link = getAlthingiLink(case)
+        case = {
+            'number': number,
+            'name': name,
+            'case_type': case_type,
+            'althingi_link': althingi_link
+        }
+        cases.append(case)
+    return cases
+
+def getAllCases(soup):
     """
-    Function (generator) to retrieve a case from þingmálalisti
-    Interesting information:
-        - Case number
-        - Case name
-        - Case type ( single letter identifier)
-
-    Output format: [case_number, case_name, case_type]
+        Takes in a soup and returns all <mál><mál> tags
     """
-    xml_logger = logging.getLogger('xmlParser')
-    xml_logger.info('starting get_cases')
-
-    cases = case_collector(parliament_session_number)
-
-    try:
-        for number, name, case_type in cases:
-            output = [number, name, case_type]
-            yield output
-    except Exception as e:
-        xml_logger.error(e.message)
-        xml_logger.info('Iteration over all parliament cases complete')
-
-
-def case_collector(parliament_session_number):
-    xml_logger = logging.getLogger('xmlParser')
-
-    link = "http://www.althingi.is/altext/xml/thingmalalisti/?lthing="
-    soup = get_xml(link + str(parliament_session_number))
+    return soup.find_all('mál')
 
-    xml_logger.info('have the case soup')
+def getCaseNumberFromCase(soup):
+    """
+        Takes in a soup with <mál><mál> as the
+        root tag and returns the málsnumer attribute
+    """
+    return soup["málsnúmer"]
 
-    case_numbers_gen = xml_helper.get_attribute_value(
-        soup, "mál", "málsnúmer")
+def getCaseName(soup):
+    """
+        Takes in a soup and finds the case name
+    """
+    return soup.find("málsheiti").string
 
-    xml_logger.info('have the case numbers generator')
+def getAlthingiLink(soup):
```

Comment:
\o/

I'm a fan of these "mini" functions
Really clarifies what one is looking for and why